updated ripple.yml

Hedde opened the pull request:
updated ripple consensus variable, ripple nolonger maintains >50% of the default UNL's validators (see: https://minivalist.cinn.app/)

nbougalis commented:
Thanks for doing this PR! A couple of comments:

1. Looking at other entries in the list, I believe the value for `consensus_distribution` should be *2* and not *0*. The title of the column is somewhat confusing.
2. Also consider updating the `public_nodes` entry; according to https://xrpcharts.ripple.com/#/topology there are 789 nodes as of this writing.
3. Lastly, you may want to consider updating the `notes`. I'd recommend:

>The default validator list can be visualized at https://minivalist.cinn.app/ but individual server operators can edit a configuration file to specify a different list.

Hedde commented:
@nbougalis thanks for the review, after reading you comments I've reached the same conclusion for point 1, I will change accordingly. Digging deeper into the columns I wanted to ask you if the tooltip (outer right) should be changed to, it currently reads;

> By default, rippled only points to a list of only Ripple-operated validator nodes (currently 16) at https://vl.ripple.com (https://github.com/ripple/rippled/blob/develop/cfg/validators-example.txt). Node operators would have to edit a config file to change this or choose their own list - hence only 1 entity owning consensus by default.

I believe that number should be **21**

nbougalis commented:
The updated note is not accurate. The majority of servers on the list are **not** operated by Ripple.

I recommend changing the note to the following:

>The default validator list can be visualized at https://minivalist.cinn.app/ but individual server operators can edit a configuration file to specify a different list.

Hedde commented:
Updated the ripple.yml notes value accordingly to suggestion. Question; is it still informative to point out the ripple.txt file validators might expose?

Hedde commented:
- Also updated the api.json file to reflect the changes (didn't notice this before)
- Changed the name data to reflect XRP not Ripple as Ripple is a company, the asset's name is XRP
- Changed asset's url to 'https://ripple.com/xrp/' (was 'https://ripple.com')

MarkusTeufelberger commented:
> # of entities in control of >50% of voting/mining power

That's a kinda bad metric for XRPL (because 50% is not a relevant threshold there) or as a measure for decentralization in general. By definition, this number can only be 1 or 0. Why/how should it be 2?

I'd rather see a `minimal # of entities that need to collude to reliably censor valid transactions that they don't like` metric (which I think this number is supposed to represent anyways, or at least how it is used atm.) - This would mean that in PoW systems like Bitcoin, one needs a 50%+ mining power majority and could just count mining pools and this number would also make more sense in PoS systems. I wanted to wait to update XRPL with more current numbers until the decentralization part has been more fully achieved (e.g. once Ripple only operates a single validator on the mainnet).

I'd rather link to https://minivalist.cinn.app/ instead of https://ripple.com/dev-blog/decentralization-strategy-update/ in `consensus_distribution_source` and I'd prefer the link to the default config to stay intact.

Hedde commented:
> That's a kinda bad metric for XRPL (because 50% is not a relevant threshold there) or as a measure for decentralization in general. By definition, this number can only be 1 or 0. Why/how should it be 2?

Currently Ripple has a 48% stake in the Default UNL, so it needs **1** more validator to break 50%. Hence minimum joined number of _entities_ that can break the 50% boundary is **2**

```
// 10 Ripple validators
(100 / 21) * 10 ≈ 48%

// 10 Ripple validators + 1 random entity
(100 / 21) * (10 + 1) ≈ 52%
```

You are right that this doesn't mean much as one needs supermajority, _which would mean an even higher number_ but this does seem the way the metric is measured as of now, so I don't think we should differentiate it based on consensus rules. Hence Bitcoin is at 3, so is Ethereum, Dash is at 4.

MarkusTeufelberger commented:
> It's 3 for both Bitcoin, Monero and Ethereum

So how can 3 entities at the same time have >50% control of voting/mining power?

Hedde commented:
It represents a cummulative metric

MarkusTeufelberger commented:
Then it is wrongly named.
